### DISCUSSION TITLE
How to differentiate releases from different branches on same Release pipeline

### DISCUSSION THREAD

**Discussion by geertbleyen** [rfennell/AzurePipelines#1550] [Q&A]
Hi,

we are using a single release pipeline to deploy our app to Apple's AppStoreConnected and MS's AppCenter. However, we are working from different branches at the same time (for example, stabilizing 1.0 and new features already on dev, which bear 2.0).
Those platforms process the uploaded packages and check their versionnumbers to handle correct version grouping etc.

Maybe I'm missing something obvious, but currently, when it generates the release notes, it does look at the previous successfully deployed release, but it can happen that chronologically the previous deployment was from a different source branch.
Is there a straightforward way to let the changelog generation taken source branch of the primary artifact into account?
Currently, I work around this issue by having cloned release definitions for dev and 1.0 branch triggers, which is undesired in context of pipeline maintenance.

Any help or tips are appreciated.

**Comment by rfennell** [comment]
The way I address problems like this is to add 'dummy stages' that are only entered on specific branches. If you then place the generate release notes in this stage (you may well end up with multiple instances of the task) and set the correct flags it will generate the release notes since the last successful  running of the task in that stage (and hence branch)

See the [wiki](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task#use-cases), you need `checkstages=true` parameter

The a video on usage of this task in this and other manners is discussed in this [DDD Community Conference Session](https://www.youtube.com/watch?v=xaV3dFoQdV8&t=331s)
